src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/XamlIlAvaloniaPropertyHelper.cs

```diff
@@ -69,6 +68,13 @@ public static IXamlIlAvaloniaPropertyNode CreateNode(AstTransformationContext co
             if(parsedPropertyName.owner == null)
                 forgedReference = new XamlAstNamePropertyReference(lineInfo, selectorTypeReference,
                     propertyName, selectorTypeReference);
+            else if (string.IsNullOrWhiteSpace(parsedPropertyName.ns)
```

Comment:
We **need** validation, to allow class bindings only in ControlThemes and Styles without any activators. This validation is likely need to happen in AvaloniaXamlIlSetterTransformer though, passing boolean or flag to the XamlIlAvaloniaPropertyHelper.

Reply:
This is needed, because currently our styles model doesn't have any protection from style/layout cycles. Disallowing any activators means we keep styles simpler and less prone to cycle error.